week3/class_version/monte_carlo_tictactoe.py

```diff
@@ -0,0 +1,172 @@
+    function should score the completed board and update the scores
+    grid. As the function updates the scores grid directly,
+    it does not return anything.
+    """
+    if not board.check_win() == provided.DRAW:
+        apply_score(scores, board, player)
+
+def get_score_values(board, player):
+    """
+    Return the values to score each players moves with
+    """
+    if player == board.check_win():
+        return SCORE_CURRENT, -SCORE_OTHER
+    else:
+        return -SCORE_CURRENT, SCORE_OTHER    
+
+def apply_score(scores, board, player):
+    """
+    Add the scores to the scores grid
+    """
+    def update_tile_scores(row, col): 
+        """
+        Update an individual tile in the scores board
+        """
+        if board.square(row, col) == player:
+            scores[row][col] += score_current
+        elif not board.square(row, col) == provided.EMPTY:
+            scores[row][col] += score_other  
+    
+    score_current, score_other = get_score_values(board, player)
+
+    [[update_tile_scores(row, col) for col in range(board.get_dim())] 
+    for row in range(board.get_dim())]
+            
+    # for row in range(board.get_dim()):
+    #     for col in range(board.get_dim()):
+            # if board.square(row, col) == player:
+            #     scores[row][col] += score_current
+            # elif not board.square(row, col) == provided.EMPTY:
+            #     scores[row][col] += score_other
+    
+# def update_tile_scores(board, scores, row, col, player, score_current, score_other): 
+#     #I don't like that I am passing in soooo many parameters
+#     #I will keep working on a refactor
+#     """
+#     Update an individual tile in the scores board
+#     """
+#     if board.square(row, col) == player:
+#         scores[row][col] += score_current
+#     elif not board.square(row, col) == provided.EMPTY:
+#         scores[row][col] += score_other   
+
+def get_best_move(board, scores):
+    """
+    This function takes a current board and a grid of scores.
+    The function should find all of the empty squares with the
+    maximum score and randomly return one of them as a (row,column)
+    tuple. It is an error to call this function with a board that
+    has no empty squares (there is no possible next move), so your
+    function may do whatever it wants in that case. The case where
+    the board is full will not be tested.
+    """
+    if len(board.get_empty_squares()) > 0:
+        high_scores = get_high_score(board, scores)
+        return select_random_highest_tile(high_scores)
+
+def get_high_score(board, scores):
+    """
+    Generates a list of the indices high scores to choose from
+    """
+    high_scores = []
+    high_index = board.get_empty_squares()[0]
+    highest_score = scores[high_index[0]][high_index[1]]
+
+    for index in board.get_empty_squares():
+        high_scores, highest_score = update_high_scores(scores[index[0]][index[1]], index, highest_score, high_scores)
```

Comment:
You pass in `index[0]` and `index[1]` and `index` itself. It'd be cleaner if you just did:
```
update_high_scores(scores, index, highest_score, high_scores)
```
and unpacked the stuff in the function.